src/msg/async/AsyncConnection.cc

```diff
@@ -362,14 +362,21 @@ ssize_t AsyncConnection::_try_send(bool more)
                              << " remaining bytes " << outcoming_bl.length() << dendl;
 
   if (!open_write && is_queued()) {
-    center->create_file_event(sd, EVENT_WRITABLE, write_handler);
-    open_write = true;
+    if (center->in_thread()) {
```

Comment:
could you elaborate why we would want to move the `{create,delete}_file_event()` calls into event thread in the commit message?